code/modules/reagents/reagent_containers.dm

```diff
@@ -116,6 +116,14 @@
 			span_userdanger("You feel drenched!"),
 		)
 
+	playsound(target, 'sound/effects/slosh.ogg', 25, TRUE)
+
+	var/image/splash_animation = image('icons/effects/effects.dmi', target, "splash")
+	if(isturf(target))
+		splash_animation = image('icons/effects/effects.dmi', target, "splash_floor")
+	splash_animation.color = mix_color_from_reagents(reagents.reagent_list)
+	flick_overlay_global(splash_animation, GLOB.clients, 0.8 SECONDS)
```

Comment:
should probably flick_overlay_view() but i'm not entirely sure

Reply:
from what I was told flick_overlay_global would do the same in this case because it only displays to those who can view the target the image is placed on, while also being slightly more efficient